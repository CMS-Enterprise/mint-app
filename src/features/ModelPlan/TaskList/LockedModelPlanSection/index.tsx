import React from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation, useParams } from 'react-router-dom';

import Breadcrumbs, { BreadcrumbItemOptions } from 'components/Breadcrumbs';
import UswdsReactLink from 'components/LinkWrapper';
import MainContent from 'components/MainContent';
import PageHeading from 'components/PageHeading';

interface LocationState {
  state: {
    error: boolean;
    route: string;
  };
  error: boolean;
  route: string;
}
const LockedModelPlanSection = () => {
  const { t } = useTranslation('modelPlanTaskList');
  const location = useLocation<LocationState>();
  const { error, route } = location?.state;
  const { modelID } = useParams<{ modelID: string }>();

  return (
    <MainContent className="mint-not-found grid-container">
      {route && (
        <Breadcrumbs
          items={[
            BreadcrumbItemOptions.HOME,
            BreadcrumbItemOptions.COLLABORATION_AREA
          ]}
          customItem={t(`breadCrumbState.${route}`)}
        />
      )}

      {!error ? (
        <div className="margin-y-7">
          <PageHeading className="margin-bottom-2" data-testid="page-locked">
            {t('lockedHeading')}
          </PageHeading>
          <p>{t('lockedSubheading')}</p>

          <UswdsReactLink
            className="usa-button margin-top-6"
            variant="unstyled"
            to={`/models/${modelID}/collaboration-area/task-list`}
          >
            {t('returnToTaskList')}
          </UswdsReactLink>
        </div>
      ) : (
        <div className="margin-y-7">
          <PageHeading className="margin-bottom-2" data-testid="page-error">
            {t('lockErrorHeading')}
          </PageHeading>
          <p>{t('lockErrorInfo')}</p>

          <UswdsReactLink
            className="usa-button margin-top-6"
            variant="unstyled"
            to={`/models/${modelID}/collaboration-area/task-list`}
          >
            {t('returnToCollaboration')}
          </UswdsReactLink>
        </div>
      )}
    </MainContent>
  );
};

export default LockedModelPlanSection;
