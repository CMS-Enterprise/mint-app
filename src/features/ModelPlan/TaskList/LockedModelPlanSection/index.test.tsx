import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { render, waitFor } from '@testing-library/react';

import LockedModelPlanSection from '.';

const modelID = '53e06174-83b9-462a-86d0-3a936f3c2670';

describe('Locked Task List Section Page', () => {
  it('renders a page that is locked', async () => {
    const { getByTestId } = render(
      <MemoryRouter
        initialEntries={[
          {
            pathname: `/models/${modelID}/locked-task-list-section`,
            state: { error: false, route: 'basics' }
          }
        ]}
      >
        <Route path="/models/:modelID/locked-task-list-section">
          <LockedModelPlanSection />
        </Route>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(getByTestId('page-locked')).toHaveTextContent(
        'Someone is currently editing the section youâ€™re trying to access.'
      );
    });
  });

  it('renders a page that is errored', async () => {
    const { getByTestId } = render(
      <MemoryRouter
        initialEntries={[
          {
            pathname: `/models/${modelID}/locked-task-list-section`,
            state: { error: true, route: 'basics' }
          }
        ]}
      >
        <Route path="/models/:modelID/locked-task-list-section">
          <LockedModelPlanSection />
        </Route>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(getByTestId('page-error')).toHaveTextContent(
        'Sorry, an has error occured.'
      );
    });
  });

  it('matches snapshot', async () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[
          {
            pathname: `/models/${modelID}/locked-task-list-section`,
            state: { error: false, route: 'basics' }
          }
        ]}
      >
        <Route path="/models/:modelID/locked-task-list-section">
          <LockedModelPlanSection />
        </Route>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
