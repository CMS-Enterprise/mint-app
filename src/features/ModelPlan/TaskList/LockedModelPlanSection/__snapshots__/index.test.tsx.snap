// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Locked Task List Section Page > matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="mint-main-content mint-not-found grid-container"
    id="main-content"
    role="main"
    tabindex="-1"
  >
    <nav
      aria-label="Breadcrumbs"
      class="usa-breadcrumb usa-breadcrumb--wrap"
    >
      <ol
        class="usa-breadcrumb__list"
      >
        <li
          class="usa-breadcrumb__list-item"
        >
          <a
            class="usa-link usa-breadcrumb__link"
            href="/"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li
          class="usa-breadcrumb__list-item"
        >
          <a
            class="usa-link usa-breadcrumb__link"
            href="/models/53e06174-83b9-462a-86d0-3a936f3c2670/collaboration-area"
          >
            <span>
              Model collaboration area
            </span>
          </a>
        </li>
        <li
          aria-current="page"
          class="usa-breadcrumb__list-item usa-current"
        >
          <span>
            Model basics
          </span>
        </li>
      </ol>
    </nav>
    <div
      class="margin-y-7"
    >
      <h1
        aria-live="polite"
        class="mint-h1 margin-bottom-2"
        data-testid="page-locked"
        tabindex="-1"
      >
        Someone is currently editing the section youâ€™re trying to access.
      </h1>
      <p>
        Please try again later.
      </p>
      <a
        class="usa-button margin-top-6"
        href="/models/53e06174-83b9-462a-86d0-3a936f3c2670/collaboration-area/task-list"
      >
        Return to the task list
      </a>
    </div>
  </main>
</DocumentFragment>
`;
